pulsar-broker/src/main/java/org/apache/pulsar/broker/resourcegroup/ResourceUsageTransportManager.java

```diff
@@ -0,0 +1,243 @@
+            resourceUsagePublishTask.cancel(true);
+            producer.close();
+        }
+    }
+
+    private class ResourceUsageReader implements ReaderListener<byte[]>, AutoCloseable {
+        private final ResourceUsageInfo recdUsageInfo = new ResourceUsageInfo();
+        private final Reader<byte[]> consumer;
+
+        public ResourceUsageReader() throws PulsarClientException {
+            consumer =  pulsarClient.newReader()
+                    .topic(pulsarService.getConfig().getResourceUsageTransportPublishTopicName())
+                    .startMessageId(MessageId.latest)
+                    .readerListener(this)
+                    .create();
+            }
+
+        @Override
+        public void close() throws Exception {
+            consumer.close();
+        }
+
+        @Override
+        public void received(Reader<byte[]> reader, Message<byte[]> msg) {
+            try {
+                recdUsageInfo.parseFrom(Unpooled.wrappedBuffer(msg.getData()), msg.getData().length);
+
+                recdUsageInfo.getUsageMapsList().forEach(ru -> {
+                    ResourceUsageConsumer owner = consumerMap.get(ru.getOwner());
+                    if (owner != null) {
+                        owner.resourceUsageListener(recdUsageInfo.getBroker(), ru);
+                    }
+                });
+
+            } catch (IllegalStateException exception) {
+                LOG.error("Resource usage reader: Error parsing incoming message {}", exception);
+            } catch (Exception exception) {
+                LOG.error("Resource usage reader: Unknown exception while parsing message {}", exception);
+            }
+        }
+
+    }
+
+    private static final Logger LOG = LoggerFactory.getLogger(ResourceUsageTransportManager.class);
+    private final PulsarService pulsarService;
+    private final PulsarClient pulsarClient;
+    private final ResourceUsageWriterTask pTask;
+    private final ResourceUsageReader consumer;
+    private final Map<String, ResourceUsagePublisher>
+            publisherMap = new ConcurrentHashMap<String, ResourceUsagePublisher>();
+    private final Map<String, ResourceUsageConsumer>
+            consumerMap = new ConcurrentHashMap<String, ResourceUsageConsumer>();
+
+    private void createTenantAndNamespace() throws PulsarServerException, PulsarAdminException {
+        // Create a public tenant and default namespace
+        TopicName topicName = TopicName.get(pulsarService.getConfig().getResourceUsageTransportPublishTopicName());
+
+        PulsarAdmin admin = pulsarService.getAdminClient();
+        ServiceConfiguration config = pulsarService.getConfig();
+        String cluster = config.getClusterName();
+
+        final String tenant = topicName.getTenant();
+        final String namespace = topicName.getNamespace();
+
+        List<String> tenantList =  admin.tenants().getTenants();
+        if (!tenantList.contains(tenant)) {
+            admin.tenants().createTenant(tenant,
```

Comment:
Every broker when attempt to run this correct?  If so, I would add some retry logic for this whole block in case multiple brokers are trying to do this at the same time and only one suceeds.

Reply:
@jerrypeng Yes, every broker will attempt to run this. I can add the retry logic, but I don't understand why we need one. One of the brokers will succeed and others will find that the tenant/namespace exist and move on. The one race condition that is not handled is if after the check for exists, the tenant gets created by some other broker, this broker might get ConflictException. I will add a try/catch to ignore this exception. However, can I please take that up in my next commit? I can create a github issue to track it.

Reply:
> The one race condition that is not handled is if after the check for exists, the tenant gets created by some other broker, this broker might get ConflictException

Yes that exact scenario. When have added similar logic in the worker:

https://github.com/apache/pulsar/blob/master/pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/PulsarWorkerService.java#L229

To handle that exact situation. It is a lot more lightweight solution that to force a restart.

> However, can I please take that up in my next commit? I can create a github issue to track it.

sure

Reply:
> @jerrypeng Yes, every broker will attempt to run this. I can add the retry logic, but I don't understand why we need one. One of the brokers will succeed and others will find that the tenant/namespace exist and move on. The one race condition that is not handled is if after the check for exists, the tenant gets created by some other broker, this broker might get ConflictException. I will add a try/catch to ignore this exception. However, can I please take that up in my next commit? I can create a github issue to track it.

To add try/catch is a good solution.